rest/resource-server/src/main/java/org/eclipse/sw360/rest/resourceserver/moderationrequest/Sw360ModerationRequestService.java

```diff
@@ -242,102 +245,93 @@ public Map<PaginationData, List<ModerationRequest>> getRequestsByState(User sw36
-                if (request.isRequestDocumentDelete()) {
-                    sw360ComponentClient.deleteComponent(request.getDocumentId(), reviewer);
-                } else {
-                    sw360ComponentClient.updateComponentFromModerationRequest(request.getComponentAdditions(),
-                            request.getComponentDeletions(), reviewer);
-                }
-                break;
+        case COMPONENT: {
+            if (request.isRequestDocumentDelete()) {
+                actionStatus = sw360ComponentClient.deleteComponent(request.getDocumentId(), reviewer);
+            } else {
+                actionStatus = sw360ComponentClient.updateComponentFromModerationRequest(
+                        request.getComponentAdditions(), request.getComponentDeletions(), reviewer);
             }
+        }
+            break;
 
-            case PROJECT: {
-                ProjectService.Iface sw360ProjectClient = getThriftProjectClient();
-                if (request.isRequestDocumentDelete()) {
-                    sw360ProjectClient.deleteProject(request.getDocumentId(), reviewer);
-                } else {
-                    sw360ProjectClient.updateProjectFromModerationRequest(request.getProjectAdditions(),
-                            request.getProjectDeletions(), reviewer);
-                }
-                break;
+        case PROJECT: {
+            if (request.isRequestDocumentDelete()) {
+                actionStatus = sw360ProjectClient.deleteProject(request.getDocumentId(), reviewer);
+            } else {
+                actionStatus = sw360ProjectClient.updateProjectFromModerationRequest(request.getProjectAdditions(),
+                        request.getProjectDeletions(), reviewer);
             }
-
-            case RELEASE: {
-                ComponentService.Iface sw360ComponentClient = getThriftComponentClient();
-                if (request.isRequestDocumentDelete()) {
-                    RequestStatus deleteStatus = sw360ComponentClient.deleteRelease(request.getDocumentId(), reviewer);
-                    if (deleteStatus.equals(RequestStatus.SUCCESS)) {
-                        SW360Utils.removeReleaseVulnerabilityRelation(request.getDocumentId(), userFromRequest);
-                    }
+        }
+            break;
+        case RELEASE: {
+            if (request.isRequestDocumentDelete()) {
+                actionStatus = sw360ComponentClient.deleteRelease(request.getDocumentId(), reviewer);
+                if (actionStatus.equals(RequestStatus.SUCCESS)) {
+                    SW360Utils.removeReleaseVulnerabilityRelation(request.getDocumentId(), userFromRequest);
                 } else {
-                    sw360ComponentClient.updateReleaseFromModerationRequest(request.getReleaseAdditions(),
-                            request.getReleaseDeletions(), reviewer);
+                    throw new TException("Failed to delete the release with ID: " + request.getDocumentId());
```

Comment:
All `actionStatus` are processed at the end of the function with `return ModerationState.REJECTED;`. Why a special `throw` here?

Also, would be better to use the `SW360Exception` instead of `TException` from thrift.